### PR TITLE
feat: added IOU token

### PR DESCRIPTION
**Pull request by hexonaut**
Token implementation is flexible. It just needs to provide `mint()` and `burn()` functions. I've used a DSToken as an example, but we could also consider modifying DSToken to give a permanent approval to `gov`.

### CODE REVIEW

**Comment by gbalabasquer** [comment]
Few comments:

- I've made `govToken` immutable, that should be merged in this branch and `iouToken` should also be immutable.
- Maybe we can add a check in some test checking the balance of the iou token after `lock` and `free`.

File: src/DssGov.t.sol [hunk 1]
```diff
@@ -78,6 +78,7 @@ contract GovUser {
     }
 
     function doFree(uint256 wad) public {
+        DSToken(address(gov.iouToken())).approve(address(gov), wad);
```
Comment (brianmcmichael) [inline src/DssGov.t.sol:81]: This approval could probably take place in the constructor and approve `uint(-1)`

Comment (hexonaut) [reply src/DssGov.t.sol:81]: Yeah the token is generic, so we could definitely use a modified `DSToken` with `DssGov` pre-approved. I just used a vanilla `DSToken` in the test. Would you prefer I add a modified `DSToken` in this PR?

Comment (gbalabasquer) [reply src/DssGov.t.sol:81]: I think you guys are referring to different stuff. I believe Brian just says in the test you can set this approval to -1 when creating the test user.
The other option is what we discussed with Sam, having a custom token that allows `dss-gov` as a powerful user, so any transferFrom coming from it doesn't require an approval at all.

Comment (brianmcmichael) [reply src/DssGov.t.sol:81]: Ah, sorry, this is the test. It's not critical to change here.

**Review by gbalabasquer — APPROVED** [review]
LGTM. Please use "squash and merge" option when getting this merged.

**Pull request merged**